GH1229 Update how-pipx-works documentation to include pyproject.toml (#1274)
pypa/pipx: docs/how-pipx-works.md

pipx install MY_PACKAGE
```

> make sure you include `project.scripts` and, optionally for Windows GUI applications, `project.gui-scripts` sections in `pyproject.toml`
> 

Reviewer: ```suggestion
make sure you include `scripts` and, optionally for Windows GUI applications, `gui-scripts` sections under your main table[^1] in `pyproject.toml`

[^1]: This is often the `[project]` table, but might also be differently named. Read more in the [PyPUG](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#writing-your-pyproject-toml). 

```

```
[project.scripts]
foo = "my_package.some_module:main_func"
bar = "other_module:some_func"

[project.gui-scripts]
baz = "my_package_gui:start_func"
```

> In this case `main_func` and `some_func` would be available to pipx after installing the above example package.

Reviewer: ```suggestion
In this case `foo` and `bar` (and `baz` on Windows), e. g. their corresponding entry point functions, would be available as "applications" to pipx after installing the above example package.
```

If you wish to provide documentation via `man` pages on UNIX-like systems then these can be added via a `tool.setuptools.data-files` section in `pyproject.toml`:

```
[tool.setuptools.data-files]
"share/man/man1" = [ "manpage.1",]
```

> In this case the manual page `manpage.1` would be available to pipx after installing the above example package.

Reviewer: ```suggestion
In this case the manual page `manpage.1` could be accessed by the user after installing the above example package.
```

Note that the `data-files` keyword is "discouraged" in the [setuptools documentation](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#setuptools-specific-configuration) but there is no alternative if `man` pages are a requirement.

> Alternatively, the legacy `setup.py` file can be used for configuration with the equivalent `entry_point` and `data_files` keys:

Reviewer: Let's drop the `setup.py` example and give a `setup.cfg` one?

pipx install MY_PACKAGE
```

make sure you include `scripts` and, optionally for Windows GUI applications, `gui-scripts` sections under your main table[^1] in `pyproject.toml`

[^1]: This is often the `[project]` table, but might also be differently named. Read more in the [PyPUG](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#writing-your-pyproject-toml).

```
[project.scripts]
foo = "my_package.some_module:main_func"
bar = "other_module:some_func"

[project.gui-scripts]
baz = "my_package_gui:start_func"
```

In this case `foo` and `bar` (and `baz` on Windows), e. g. their corresponding entry point functions, would be available as "applications" to pipx after installing the above example package.

If you wish to provide documentation via `man` pages on UNIX-like systems then these can be added via a `tool.setuptools.data-files` section in `pyproject.toml`:

```
[tool.setuptools.data-files]
"share/man/man1" = [ "manpage.1",]
```

In this case the manual page `manpage.1` could be accessed by the user after installing the above example package.

> Note that the `data-files` keyword is "discouraged" in the [setuptools documentation](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#setuptools-specific-configuration) but there is no alternative if `man` pages are a requirement.

Reviewer 2: I'd suggest putting this into a note block using [admonitions](https://squidfunk.github.io/mkdocs-material/reference/admonitions/).

Author: OK, I did that, following the style of a warning in the README.md. Since I needed to run `nox` to check the docs built as expected I decided to make the code samples tabbed in the style of other PyPA documentation. This is in a separate commit if you don't like it:

<img width="294" alt="image" src="https://github.com/pypa/pipx/assets/3342745/150575ba-6091-415d-b50f-007e4414a651">

Reviewer 2: Looks great, thanks :)
